mobile/src/screens/HomeScreen.tsx

```diff
@@ -19,6 +19,19 @@ export const HomeScreen: React.FC = () => {
     .slice(0, 5);
   const allBooks = booksCtx.books;
 
+  let newBooksRender;
+  let colBooksRender;
+  const booksLoaded: boolean = newBooks.length > 0;
+
+  // only load books if array is non-empty. Else load acitivty indicator
+  if (booksLoaded) {
+    newBooksRender = <HorizontalBookList books={newBooks} />;
+    colBooksRender = <ColumnBookList books={allBooks} />;
+  } else {
+    newBooksRender = <ActivityIndicator/>;
+    colBooksRender = <ActivityIndicator/>;
+  }
```

Comment:
This isn't bad but if you want to make it a little less verbose you can just do:

```
{ booksCtx.loading && <ActivityIndicator/> }
{ !booksCtx.loading && <HorizontalBookList books={newBooks} }
```

within your JSX.